vm/string.go

```diff
@@ -0,0 +1,11 @@
+package vm
+
+import "C"
+import "unsafe"
+
+// cstring creates a null-terminated C string from the given byte slice.
+// the caller is responsible for freeing the underlying memory
+func cstring(data []byte) *C.char {
+	str := unsafe.String(unsafe.SliceData(data), len(data))
```

Comment:
Should we do https://github.com/golang/go/blob/a03bdc3e6bea34abd5077205371e6fb9ef354481/src/runtime/string.go#L136C9-L136C39 instead?

because `len(data)` is not always equal to `len(str)` with multi-byte characters right?

Reply:
Actually `len(str)` equals `len([]byte(str))`, here is spec - https://go.dev/ref/spec#:~:text=a%20particular%20architecture.-,String%20types,-A%20string%20type

I double checked it with this test:
```
func TestString(t *testing.T) {
	str := "hello 👋"
	bytes := []byte(str)

	var runesCount int
	for range str {
		runesCount++
	}

	fmt.Printf("Str len %d, bytes len %d, runes count %d\n", len(str), len(bytes), runesCount)
}
```

I know that is one of confusing parts of string type in Go :)